weed/worker/tasks/iceberg/compact.go

```
@@ -1149,19 +1150,48 @@ func mergeParquetFilesSorted(
+		),
+	)
+	// Closing returns the run files to the pool, which deletes them. A failure
+	// before that has to release them too, or the job leaves temp files behind.
+	closed := false
+	defer func() {
+		if !closed {
+			writer.Reset(io.Discard)
+		}
+	}()
+
+	writeRows := func(reader *parquet.Reader, source string) (int64, error) {
```

golangci-lint (2.13.2)

[error] SA1019: github.com/parquet-go/parquet-go.Reader is deprecated: A Reader reads Go values from parquet files.

(staticcheck)